code/game/objects/systems/cell_slot.dm

```diff
@@ -48,6 +54,21 @@
 	cell = null
 	parent.object_cell_slot_removed(., src)
 
+/**
+ * removes cell from the system and drops it in users hands
+ */
+/datum/object_system/cell_slot/proc/mob_remove_cell(mob/user)
+	if(isnull(cell))
+		return
+	. = cell
+	user.put_in_hands_or_drop(cell)
+	cell = null
+	parent.object_cell_slot_removed(., src)
+	return TRUE
```

Comment:
this entire thing should be using remove_cell() instead of doing it a second time
duplicate code is less maintainable

also decide if it'll either
1. return TRUE / FALSE for success / failure
2. better, return cell if success, null if failure

and document as such please

Reply:
Replaced with silicons code